hyppo/d_variate/dhsic.py

```diff
@@ -0,0 +1,147 @@
+from .base import DVariateTest, DVariateTestOutput
+from ..tools import multi_compute_kern
+from ._utils import _CheckInputs
+
+import numpy as np
+
+
+class dHsic(DVariateTest):
+    r"""
+    d-variate Hilbert Schmidt Independence Criterion (dHsic) test statistic
+    and p-value.
+
+    dHsic is a non-parametric kernel-based independence test between an
+    arbitrary number of variables. The dHsic statistic is 0 if the variables
+    are jointly independent and positive if the variables are dependent.
+    :footcite:p:`grettonKernelJointIndependence2016`.
+    The default choice is the Gaussian kernel, which uses the median distance
+    as the bandwidth, which is a characteristic kernel that guarantees that
+    dHsic is a consistent test.
+    :footcite:p:`grettonKernelStatisticalTest2007`
+    :footcite:p:`grettonConsistentNonparametricTests2010`
+    :footcite:p:`grettonKernelJointIndependence2016`.
+
+    Parameters
+    ----------
+    compute_kernel : str, callable, or None, default: "gaussian"
+        A function that computes the kernel similarity among the samples within each
+        data matrix.
+        Valid strings for ``compute_kernel`` are, as defined in
+        :func:`sklearn.metrics.pairwise.pairwise_kernels`,
+
+            [``"additive_chi2"``, ``"chi2"``, ``"linear"``, ``"poly"``,
+            ``"polynomial"``, ``"rbf"``,
+            ``"laplacian"``, ``"sigmoid"``, ``"cosine"``]
+
+        Note ``"rbf"`` and ``"gaussian"`` are the same metric.
+        Set to ``None`` or ``"precomputed"`` if ``args`` are already similarity
+        matrices. To call a custom function, either create the similarity matrix
+        before-hand or create a function of the form :func:`metric(x, **kwargs)`
+        where ``x`` is the data matrix for which pairwise kernel similarity matrices are
+        calculated and kwargs are extra arguments to send to your custom function.
+    bias : bool, default: False
+        Whether or not to use the biased or unbiased test statistics.
+    **kwargs
+        Arbitrary keyword arguments for ``multi_compute_kern``.
+
+    Notes
+    -----
+    The statistic can be derived as follows
+    :footcite:p:`grettonKernelJointIndependence2016`:
+
+    dHsic builds on the two-variable Hilbert Schmidt Independence Criterion (Hsic),
+    implemented in :class:`hyppo.independence.Hsic`, but allows for an arbitrary
+    number of variables. For a given kernel, the joint distribution and the product
+    of the marginals is mapped to the reproducing kernel Hilbert space and the
+    squared distance between the embeddings is calculated. The dHsic statistic can
+    be calculated by,
+
+    .. math::
+
+        \mathrm{dHsic} (\mathbb{P}^{(X^1, ..., X^d)}) = \Vert \prod(\mathbb{P}^{X^1}
```

Comment:
Recheck this tex, renders weird on the website: https://deploy-preview-233--hyppo.netlify.app/api/generated/hyppo.d_variate.dhsic#hyppo.d_variate.dHsic